src/rest_api/api/search/identity_search.rs

```diff
@@ -79,30 +77,27 @@ pub fn v1_index_search_identity(
         extra_query,
     );
 
-    let mut results: HashSet<DocAddress> = HashSet::with_capacity(1);
-
     match result {
         Ok(Some(result)) => {
-            results.insert(result);
-        }
-        Ok(None) => return GetStructureSearchResponse::Ok(Json(Vec::new())),
-        Err(e) => {
-            return GetStructureSearchResponse::Err(Json(StructureResponseError {
-                error: e.to_string(),
-            }))
-        }
-    };
+            let schema = searcher.schema();
+            let smiles_field = schema.get_field("smiles").unwrap();
+            let extra_data_field = schema.get_field("extra_data").unwrap();
 
-    let final_results = aggregate_search_hits(searcher, results, false, &smiles);
+            let (smiles, extra_data) =
+                get_smiles_and_extra_data(result, &searcher, smiles_field, extra_data_field)
+                    .unwrap();
```

Comment:
I am concerned about an `unwrap()` in an API call but I'm going to let it slide for now 😏